### PR TITLE
Update website to Node 16

### PR DESCRIPTION
**Pull request by danstepanov**
# Why

We want all of snack to be using Node 16 over 12

# How

* [website] Update CI to use Node 16

* [website] Bump to Node 16.14.2

* [website] Remove TS reference to snack-sdk and snack-content

* [website] Update skaffold to reflect node 16.14.2

# Test Plan

TBD

### CODE REVIEW

File: website/package.json [hunk 1]
```diff
@@ -76,6 +77,7 @@
     "recast": "^0.20.3",
     "redux": "^4.0.1",
     "sanitize-html": "^1.20.0",
+    "snack-content": "*",
```
[the remark this answers is not in the document]
Comment (danstepanov) [reply website/package.json:80]: I'll also remove this from CI then

**Comment by byCedric** [comment]
Looking good. Almost ready to merge! Just the TS issues reported in CI 😄

File: website/src/client/components/Import/ImportFilesManager.tsx [hunk 2]
```diff
@@ -136,7 +138,7 @@ export default class ImportFilesManager extends React.PureComponent<Props, State
         folderMappings[item.name] = getUniquePath(allPaths, item.name);
       }
     });
-
+    // @ts-ignore
```
Comment (byCedric) [inline website/src/client/components/Import/ImportFilesManager.tsx:141]: What's the error you get for this one?

File: website/src/client/components/Import/ImportFilesManager.tsx [hunk 3]
```diff
@@ -87,11 +87,13 @@ export default class ImportFilesManager extends React.PureComponent<Props, State
...
-          const file = Array.from(dataTransfer.files).find((file) => file.name === entry.name);
-          return entry.isFile && file ? file : entry;
-        });
+        entries = Array.from(dataTransfer.items)
+          .map((it) => it.webkitGetAsEntry())
+          .filter((entry): entry is globalThis.FileSystemEntry => entry != null)
+          .map((entry) => {
+            const file = Array.from(dataTransfer.files).find((file) => file.name === entry.name);
+            return entry.isFile && file ? file : entry;
+          });
```
Comment (byCedric) [inline website/src/client/components/Import/ImportFilesManager.tsx:96]: This makes sense, not sure in what cases `it.webkitGetAsEntry` would return null though. The only thing I would change is this:

```suggestion
        entries = Array.from(dataTransfer.items)
          .map((it) => it.webkitGetAsEntry())
          .filter((entry): entry is FileSystemEntry => entry != null)
          .map((entry) => {
            const file = Array.from(dataTransfer.files).find((file) => file.name === entry.name);
            return entry.isFile && file ? file : entry;
          });
```

File: website/src/client/components/Import/ImportFilesManager.tsx [hunk 4]
```diff
@@ -21,7 +21,7 @@ type Props = {
   className?: string;
 };
 
-type FileItem = File | WebkitFileEntry | WebkitDirectoryEntry;
+type FileItem = File | WebkitFileEntry | WebkitDirectoryEntry | globalThis.FileSystemEntry;
```
Comment (byCedric) [inline website/src/client/components/Import/ImportFilesManager.tsx:24]: I don't think this one is required, right? The entries aren't `FileItem`s or `FileSystemEntry`'s, it's something that's being used after everything is processed.

Comment (danstepanov) [reply website/src/client/components/Import/ImportFilesManager.tsx:24]: without this added type, there is a typescipt error [here](https://github.com/expo/snack/blob/ad8634a384f5f586af2ddf89effa19483dbc1cb3/website/src/client/components/Import/ImportFilesManager.tsx#L90). The array has both `File` and `FileSystemEntry` items.
Screen Shot 2022-04-15 at 3 56 33 PM

File: website/src/client/components/Import/ImportFilesManager.tsx (hunk 3 above)
Comment (danstepanov) [reply website/src/client/components/Import/ImportFilesManager.tsx:96]: it seems like `globalThis.FileSystemEntry` and `FileSystemEntry` may be different types. when making your requested change, typescript complains about type `FileSystemEntry` not being assignable to `FileSystemEntry | null`. I'm also not sure why this value might be null.

File: website/src/client/components/Import/ImportFilesManager.tsx (hunk 2 above)
Comment (danstepanov) [reply website/src/client/components/Import/ImportFilesManager.tsx:141]: Screen Shot 2022-04-15 at 4 03 02 PM

**Review by byCedric — APPROVED** [review]
Alright, changes make sense 😄 thanks for the clarifications

**Pull request merged**